testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/userdata/QASettingsActivity.kt

```diff
@@ -0,0 +1,108 @@
+import com.rakuten.tech.mobile.miniapp.testapp.databinding.QaSettingsActivityBinding
+import com.rakuten.tech.mobile.testapp.helper.parseDateToString
+import com.rakuten.tech.mobile.testapp.helper.parseStringToDate
+import com.rakuten.tech.mobile.testapp.ui.base.BaseActivity
+import com.rakuten.tech.mobile.testapp.ui.settings.AppSettings
+import java.util.*
+
+class QASettingsActivity : BaseActivity() {
+
+    private lateinit var settings: AppSettings
+    private lateinit var binding: QaSettingsActivityBinding
+    private var accessTokenErrorCacheData: MiniAppAccessTokenError? = null
+
+    companion object {
+        fun start(activity: Activity) {
+            activity.startActivity(Intent(activity, QASettingsActivity::class.java))
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        settings = AppSettings.instance
+        accessTokenErrorCacheData = settings.accessTokenError
+        showBackIcon()
+        binding = DataBindingUtil.setContentView(this, R.layout.qa_settings_activity)
+        binding.activity = this
+        renderScreen()
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu): Boolean {
+        menuInflater.inflate(R.menu.settings_menu, menu)
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            android.R.id.home -> {
+                finish()
+                return true
+            }
+            R.id.settings_menu_save -> {
+                update()
+                return true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    private fun renderScreen() {
+        if(accessTokenErrorCacheData != null){
+            // set up initial state.
+            when {
+                accessTokenErrorCacheData?.type != null -> {
+                    binding.switchAuthFailure.isChecked = true
+                    binding.switchOtherError.isChecked = false
+                    binding.edtCustomErrorMessage.setText(accessTokenErrorCacheData?.message ?: "")
+                }
+                else -> {
+                    binding.switchAuthFailure.isChecked = false
+                    binding.switchOtherError.isChecked = true
+                    binding.edtCustomErrorMessage.setText(accessTokenErrorCacheData?.message ?: "")
+                }
+            }
+        }else{
+            // default state
+            binding.switchAuthFailure.isChecked = false
+            binding.switchOtherError.isChecked = false
+            binding.edtCustomErrorMessage.text.clear()
+        }
+    }
+
+    private fun update() {
+        // If Authorization failure checked then authorizationFailureError type will send.
+        // If Unknown failure checked then custom type will send.
+        // If Authorization and Unknown both checked then authorizationFailureError type will send.
```

Comment:
I don't think we should allow 2 choices at the same time. That's why I suggest you either write an algorithm to uncheck other switches when you select one, or you just use a radio group with radio buttons

Reply:
@Climbatize thanks for the suggestions, i wrote an algorithm to uncheck other switches, switch looks good to me because it goes with the design with android as you mentioned switches are already being used in the  settings page.😁